src/cmd/opslevel.go

```diff
@@ -0,0 +1,69 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/alecthomas/jsonschema"
+	"github.com/opslevel/opslevel-go"
+	"github.com/spf13/cobra"
+)
+
+type OpsLevelTag struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type OpsLevelTool struct {
+	Name        string                `json:"name"`
+	Category    opslevel.ToolCategory `json:"category"`
+	Url         string                `json:"url"`
+	Environment string                `json:"environment,omitempty"`
+}
+
+type OpsLevelRepository struct {
+	Name     string `json:"name"`
+	Path     string `json:"path,omitempty"`
+	Provider string `json:"provider"` // TODO: this should be an enum
+}
+
+type OpsLevelServiceDependancy struct {
+	Alias string `json:"alias"`
+}
+
+type OpsLevelService struct {
+	Name         string                      `json:"name"`
+	Description  string                      `json:"description,omitempty"`
+	Owner        string                      `json:"owner,omitempty"`
+	Lifecycle    string                      `json:"lifecycle,omitempty"`
+	Tier         string                      `json:"tier,omitempty"`
+	Product      string                      `json:"product,omitempty"`
+	Language     string                      `json:"language,omitempty"`
+	Framework    string                      `json:"framework,omitempty"`
+	Aliases      []string                    `json:"aliases,omitempty"`
+	Tags         []OpsLevelTag               `json:"tags,omitempty"`
+	Tools        []OpsLevelTool              `json:"tools,omitempty"`
+	Repositories []OpsLevelRepository        `json:"repositories,omitempty"`
+	Dependencies []OpsLevelServiceDependancy `json:"dependencies,omitempty"`
+}
+
+type OpsLevelConfig struct {
+	Version int             `json:"version"`
+	Service OpsLevelService `json:"service"`
```

Comment:
@maxim-opslevel i think we just add this here and it builds 1 schema that works for both
```
Repository OpsLevelRepository `json:"repository"`
```

But i guess that question becomes - "can you define both a repository and a service inside 1 Git repo inside the same opslevel.yml?"

Reply:
Added "one_of" requirement